ledger/ledgercore/statedelta.go

```diff
@@ -149,3 +152,31 @@ func (ad *AccountDeltas) upsert(br basics.BalanceRecord) {
 	}
 	ad.acctsCache[addr] = last
 }
+
+func (sd *StateDelta) Compress() {
+	sd.Accts.accts = sd.Accts.accts[:len(sd.Accts.accts)]
```

Comment:
this line is a no-op. The allocation for an array is the cap of the array, so just re-slicing it won't do much.
If you really want to reallocate it, you should create another array, and copy the data.